### ISSUE TITLE
Implement FT-Transformer Encoder with unit tests

### ISSUE THREAD

**Issue by takeshimx** [takeshimx/diamond-lens#82]
## Summary
FT-Transformer Encoder モジュールを新規作成し、自己教師あり学習で特徴量の埋め込みベクトルを生成できるようにする。

## Tasks
- [ ] `torch` を `backend/requirements.txt` に追加
- [ ] `backend/app/services/ft_transformer.py` を新規作成
  - `FTTransformerEncoder(nn.Module)` — Feature Tokenizer + [CLS] token + Transformer Encoder + Projection + Decoder
  - `train_ft_transformer()` — MSE reconstruction loss による自己教師あり学習
- [ ] `backend/tests/test_ft_transformer.py` を新規作成
  - encode output shape, forward dual output, different feature counts (3/4), training convergence

## Architecture
```
Input features → Feature Tokenizer (Linear per feature → d_model)
  → [CLS] token → Transformer Encoder (Self-Attention)
  → [CLS] output → Projection → embedding_dim vector
  → Decoder (reconstruction, training only)
```

## Files
- `backend/requirements.txt`
- `backend/app/services/ft_transformer.py` (NEW)
- `backend/tests/test_ft_transformer.py` (NEW)

**Comment by takeshimx** [comment]
Done.
